encodings/dict/src/compute/compare.rs

```diff
@@ -28,3 +33,70 @@ impl CompareFn<DictArray> for DictEncoding {
         Ok(None)
     }
 }
+
+fn compare_by_value(
+    lhs: &DictArray,
+    rhs: Scalar,
+    operator: Operator,
+) -> VortexResult<Option<ArrayData>> {
+    assert!(matches!(operator, Operator::Eq));
+    // If the RHS is constant, then we just need to compare against our encoded values.
+    let compare_result = compare(
+        lhs.values(),
+        ConstantArray::new(rhs, lhs.values().len()),
+        operator,
+    )?;
+
+    let bool = compare_result.into_bool()?;
+
+    // Couldn't find a value match, so the result is all false
+    let Some(code) = bool.boolean_buffer().set_indices().next() else {
+        return Ok(Some(
+            ConstantArray::new(false, lhs.codes().len()).into_array(),
+        ));
+    };
+
+    // Ensure the other is the same length as the dictionary
+    let compare_result = compare(
+        lhs.codes(),
+        try_cast(
+            ConstantArray::new(code, lhs.codes().len()),
+            lhs.codes().dtype(),
+        )?,
+        operator,
+    )?;
+    Ok(Some(compare_result))
```

Comment:
Similarly, you may need to cast the result to the correct nullability

Reply:
Codes now contain nullability